dspace-rest/src/main/java/org/dspace/rest/filter/ItemFilterUtil.java

```diff
@@ -0,0 +1,263 @@
+		return ConfigurationManager.getProperty("rest", "rest-report-mime-document-image");
+	}
+
+	static int countOriginalBitstream(Item item) {
+		return countBitstream(BundleName.ORIGINAL, item);
+	}
+	static int countBitstream(BundleName bundleName, Item item) {
+		int count = 0;
+        for(Bundle bundle: item.getBundles()){
+        	if (!bundle.getName().equals(bundleName.name())){
+        		continue;
+        	}
+        	count += bundle.getBitstreams().size();
+        }
+
+		return count;
+	}
+
+	static List<String> getBitstreamNames(BundleName bundleName, Item item) {
+		ArrayList<String> names = new ArrayList<String>();
+        for(Bundle bundle: item.getBundles()){
+        	if (!bundle.getName().equals(bundleName.name())){
+        		continue;
+        	}
+        	for(Bitstream bit: bundle.getBitstreams()) {
+        		names.add(bit.getName());
+        	}
+        }
+		return names;
+	}
+
+	
+	static int countOriginalBitstreamMime(Context context, Item item, String mimeList) {
+		return countBitstreamMime(context, BundleName.ORIGINAL, item, mimeList);
+	}
+	static int countBitstreamMime(Context context, BundleName bundleName, Item item, String mimeList) {
+		int count = 0;
+        for(Bundle bundle: item.getBundles()){
+        	if (!bundle.getName().equals(bundleName.name())){
+        		continue;
+        	}
+        	for(Bitstream bit: bundle.getBitstreams()) {
+            	for(String mime: mimeList.split(",")) {
+                    try {
+						if (bit.getFormat(context).getMIMEType().equals(mime)) {
+							count++;
+						}
+					} catch (SQLException e) {
+						log.error("Get format error for bitstream " + bit.getName());
+					}
+            	}
+            }
+        }
+		return count;
+	}
+
+	static int countBitstreamByDesc(BundleName bundleName, Item item, String descList) {
+		int count = 0;
+        for(Bundle bundle: item.getBundles()){
+        	if (!bundle.getName().equals(bundleName.name())){
+        		continue;
+        	}
+        	for(Bitstream bit: bundle.getBitstreams()) {
+            	for(String desc: descList.split(",")) {
+            		String bitDesc = bit.getDescription();
+            		if (bitDesc == null) {
+            			continue;
+            		}
+                    if (bitDesc.equals(desc)) {
+                    	count++;
+                    }
+            	}
+            }
+        }
+		return count;
+	}
+
+	static int countBitstreamSmallerThanMinSize(Context context, BundleName bundleName, Item item, String mimeList, String prop) {
+		long size = ConfigurationManager.getLongProperty("rest", prop);
+		int count = 0;
+        try {
+            for(Bundle bundle: item.getBundles()){
+            	if (!bundle.getName().equals(bundleName.name())){
+            		continue;
+            	}
+            	for(Bitstream bit: bundle.getBitstreams()) {
+                	for(String mime: mimeList.split(",")) {
+                        if (bit.getFormat(context).getMIMEType().equals(mime)) {
```

Comment:
Thank you for the suggestion. I will add a trim in all of the places where a string is being split.